keras_cv/models/stable_diffusion/stable_diffusion.py

```diff
@@ -640,3 +941,28 @@ def diffusion_model(self):
             if self.jit_compile:
                 self._diffusion_model.compile(jit_compile=True)
         return self._diffusion_model
+
+    @property
+    def diffusion_model_ptp(self):
+        """diffusion_model_ptp returns the diffusion model with modifications for the Prompt-to-Prompt method.
+
+        Reference:
+
+        - "Prompt-to-Prompt Image Editing with Cross-Attention Control."
+        Amir Hertz, Ron Mokady, Jay Tenenbaum, Kfir Aberman, Yael Pritch, Daniel Cohen-Or.
+        https://arxiv.org/abs/2208.01626
+        """
+        if self._diffusion_model_ptp is None:
+            if self._diffusion_model is None:
+                self._diffusion_model_ptp = self.diffusion_model
```

Comment:
I think we need to actually clone the diffusion model if we do this, as this reference will remain the same across assignments adn we will be modifying the diffusion model for all workflows

Reply:
You are correct @LukeWood, I missed this while implementing. Thanks for catching this up!
I feel that cloning the model costs extra memory and computation time and to overwrite the forward call and add control variables I needed to recompile the model to put these new changes to the ptp diffusion model.

What should be the best alternative in order to not compromise future and other workflows?